File: UI/src/lib/types/shared-link.ts
import type { Label } from './label';
import type { StatusInfo, IssuePriority } from './issue';

export interface SharedLink {
	id: string;
	token: string;
	workspace_id: string;
	created_by: string;
	scope: 'workspace' | 'team' | 'project' | 'view';
	scope_id?: string;
	filters: Record<string, string>;
	include_description: boolean;
	is_active: boolean;
	expires_at?: string;
	url: string;
	created_at: string;
	updated_at: string;
}

export interface CreateSharedLinkRequest {
	scope: 'workspace' | 'team' | 'project' | 'view';
	scope_id?: string;
	filters?: Record<string, string>;
	include_description?: boolean;
	expires_at?: string;
}

export interface UpdateSharedLinkRequest {
	is_active?: boolean;
	include_description?: boolean;
	expires_at?: string;
}

export interface PublicShareMeta {
	scope: string;
	scope_id?: string;
	scope_name: string;
	workspace_name: string;
	filters: Record<string, string>;
	statuses?: PublicStatus[];
}

export interface PublicStatus {
	id: string;
	name: string;
	category: string;
	color: string | null;
	position: number;
}

export interface PublicIssue {
	identifier: string;
	title: string;
	description?: string | null;
	status: string;
	status_info?: StatusInfo;
	priority: IssuePriority;
	labels?: Label[];
	assignees?: PublicUser[];
	due_date?: string;
	estimate?: number;
	created_at: string;
	updated_at: string;
}

export interface PublicUser {
	name: string;
	display_name: string;
	avatar_url?: string;
}

File: BE/internal/domain/shared_link.go
package domain

import (
	"encoding/json"
	"time"

	"github.com/google/uuid"
)

type SharedLinkScope string

const (
	SharedLinkScopeWorkspace SharedLinkScope = "workspace"
	SharedLinkScopeTeam      SharedLinkScope = "team"
	SharedLinkScopeProject   SharedLinkScope = "project"
	SharedLinkScopeView      SharedLinkScope = "view"
)

type SharedLink struct {
	ID                 uuid.UUID       `json:"id" db:"id"`
	Token              string          `json:"token" db:"token"`
	WorkspaceID        uuid.UUID       `json:"workspace_id" db:"workspace_id"`
	CreatedBy          uuid.UUID       `json:"created_by" db:"created_by"`
	Scope              SharedLinkScope `json:"scope" db:"scope"`
	ScopeID            *uuid.UUID      `json:"scope_id" db:"scope_id"`
	Filters            json.RawMessage `json:"filters" db:"filters"`
	IncludeDescription bool            `json:"include_description" db:"include_description"`
	IsActive           bool            `json:"is_active" db:"is_active"`
	ExpiresAt          *time.Time      `json:"expires_at" db:"expires_at"`
	CreatedAt          time.Time       `json:"created_at" db:"created_at"`
	UpdatedAt          time.Time       `json:"updated_at" db:"updated_at"`
}
